server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -237,7 +237,9 @@ private void dispatchRequest(RestRequest request, RestChannel channel, RestHandl
                 sendContentTypeErrorMessage(request.getAllHeaderValues("Content-Type"), channel);
                 return;
             }
-            if (handler.supportsContentStream() && xContentType != XContentType.JSON && xContentType != XContentType.SMILE) {
+            //todo consider moving to XContentType
```

Comment:
nit: if want to leave this todo, can you make it TODO:  and add a possible reason ... i guess i don't see how would belong in XContentType ?

Reply:
hm when thinking about this now, it would probably more fit into handler.supportsContentStream(xContentType). 
Will fix the todo